pkg/services/sqlstore/team.go

```diff
@@ -188,14 +200,14 @@ func (ss *SQLStore) SearchTeams(ctx context.Context, query *models.SearchTeamsQu
 	params := make([]interface{}, 0)
 
 	filteredUsers := getFilteredUsers(query.SignedInUser, query.HiddenUsers)
-	sql.WriteString(getTeamSelectSQLBase(filteredUsers))
-
 	for _, user := range filteredUsers {
 		params = append(params, user)
 	}
 
-	if query.UserIdFilter != models.FilterIgnoreUser {
-		sql.WriteString(` INNER JOIN team_member ON team.id = team_member.team_id AND team_member.user_id = ?`)
+	if query.UserIdFilter == models.FilterIgnoreUser {
+		sql.WriteString(getTeamSelectSQLBase(filteredUsers))
+	} else {
+		sql.WriteString(getTeamSelectWithPermissionsSQLBase(filteredUsers))
```

Comment:
One option to keep things simple here may be to change to a left join instead of an inner join, then we can have the join and the permission column in both cases and just add a where condition on `team_member.user_id=?` if we need to show only those teams the user is a member of.  That would avoid the need for 2 different base queries.